File: src/main/java/my/maryvk/maryvkweb/service/RelationChangeServiceImpl.java
package my.maryvk.maryvkweb.service;

import lombok.extern.java.Log;
import my.maryvk.maryvkweb.domain.RelationChange;
import my.maryvk.maryvkweb.repository.RelationChangeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@Log
public class RelationChangeServiceImpl implements RelationChangeService {

    private final RelationChangeRepository relationChangeRepository;

    @Autowired
    public RelationChangeServiceImpl(RelationChangeRepository relationChangeRepository) {
        this.relationChangeRepository = relationChangeRepository;
    }

    @Override
    public void registerChange(RelationChange relationChange) {
        relationChangeRepository.saveAndFlush(relationChange);
    }

    @Override
    public List<RelationChange> findAllByOwnerIdOrderByTimeDesc(int ownerId) {
        return relationChangeRepository.findAllByOwnerIdOrderByTimeDesc(ownerId);
    }

    @Override
    public List<RelationChange> findAllOrderByTimeDesc() {
        return relationChangeRepository.findAllByOrderByTimeDesc();
    }
}

File: src/main/java/my/maryvk/maryvkweb/service/RelationChangeServiceImpl.kt
package my.maryvk.maryvkweb.service

import my.maryvk.maryvkweb.domain.RelationChange
import my.maryvk.maryvkweb.repository.RelationChangeRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class RelationChangeServiceImpl
@Autowired constructor(private val relationChangeRepository: RelationChangeRepository) : RelationChangeService {

    override fun registerChange(relationChange: RelationChange) {
        relationChangeRepository.saveAndFlush(relationChange)
    }

    override fun findAllByOwnerIdOrderByTimeDesc(ownerId: Int): List<RelationChange> {
        return relationChangeRepository.findAllByOwnerIdOrderByTimeDesc(ownerId)
    }

    override fun findAllOrderByTimeDesc(): List<RelationChange> {
        return relationChangeRepository.findAllByOrderByTimeDesc()
    }
}